ld/devices.data

```diff
@@ -259,6 +259,18 @@ stm32g4?[34]?b* stm32g4ccm ROM=128K RAM=96K CCM=32K
 stm32g4?[34]?c* stm32g4ccm ROM=256K RAM=96K CCM=32K
 stm32g4?[34]?e* stm32g4ccm ROM=512K RAM=96K CCM=32K
 
+# For STM32H7 CCM is called D-TCM
+stm32h72[35]?e* stm32h7xxxe RAM=128K RAM1=16K RAM2=16K RAM4=16K CCM=128K
+stm32h72[35]?g* stm32h7xxxg RAM=128K RAM1=16K RAM2=16K RAM4=16K CCM=128K
+stm32h730?b* stm32h7xxxb RAM=128K RAM1=16K RAM2=16K RAM4=16K CCM=128K
+stm32h73[35]?g* stm32h7xxxg RAM=128K RAM1=16K RAM2=16K RAM4=16K CCM=128K
+stm32h742?i* stm32h7xxxi RAM=384K RAM1=32K RAM2=16K RAM4=64K CCM=128K
+stm32h742?g* stm32h74xxg RAM=384K RAM1=32K RAM2=16K RAM4=64K CCM=128K
+stm32h74[357]?i* stm32h7xxxi RAM=512K RAM1=128K RAM2=128K RAM3=32K RAM4=64K CCM=128K
+stm32h74[357]?g* stm32h74xxg RAM=512K RAM1=128K RAM2=128K RAM3=32K RAM4=64K CCM=128K
+stm32h75[357]?i* stm32h7xxxi RAM=512K RAM1=128K RAM2=128K RAM3=32K RAM4=64K CCM=128K
+stm32h750?b* stm32h7xxxb RAM=512K RAM1=128K RAM2=128K RAM3=32K RAM4=64K CCM=128K
```

Comment:
What's the story here with ram and ram1?  in other parts we have ram, and ram2?  is ram1 meant to be the same as ram? or can we just start at ram2 like other parts?  I know this gets weird when they can be reached contiguously, but what's the intention  here?  I'm really a bit unsure about having a second "ram1"

Reply:
RAM1 is being used here, because H7 parts specify up to 5 different RAM regions, which unfortunately are not accessible continuously. "RAM1" designator has been used with different parts already. I've used it, so I can cover all 5 non-continuous regions of H7 chips without changing the linker script.

Same applies for ROM and ROM1, while certain H7 parts contain double-bank flash, which does not need to be continuous.

Reply:
Ok, I still have to update linker script. Somehow I missed that RAM4 region is not supported there.